core/DataAccess/ArchiveWriter.php

```diff
@@ -215,8 +215,8 @@ private function batchInsertSpool($valueType)
 
         $valueSeen = false;
         foreach ($records as $record) {
-            // don't record zero
-            if (empty($record[1])) {
+            // don't record zero unless it's nb_profilable
+            if (empty($record[1]) && $record[0] != 'nb_profilable') {
```

Comment:
I know this is currently the only case for this special handling, but I was wondering if can get this into a more global solution. So not defining in the archive writer, which records to store even if empty, but maybe define them in the archiver or alternatively using an event, that can be used in plugin classes or similar. What do you think?

Reply:
I don't think it's necessary right now amd nb_profilable is a core metric with (in other prs) tight integration to core UI logic. If another plugin needs it, it would be easy to add a mechanism, but right it would mean expanding the API for something no one currently needs.